examples/admission_controller.rs

```diff
@@ -0,0 +1,93 @@
+use std::convert::{Infallible, TryInto};
+
+use kube::api::{AdmissionRequest, AdmissionResponse, AdmissionReview, DynamicObject, PatchType};
+#[macro_use] extern crate log;
+use warp::{reply, Filter, Reply};
+
+#[tokio::main]
+async fn main() {
+    std::env::set_var("RUST_LOG", "debug,kube=debug");
+    tracing_subscriber::fmt::init();
+
+    let routes = warp::path("mutate")
+        .and(warp::body::json())
+        .and_then(mutate)
+        .with(warp::trace::request());
+
+
+    // For admission controllers running behind a service, you must generate a
+    // certificate for the service and encode the CA in the
+    // MutatingWebhookConfiguration, then terminate TLS here. For controllers
+    // running outside of the cluster, while still necessary, it's fine to
+    // handle TLS upstream. See:
+    // https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/#tls-certificates
+
+    // warp::serve(warp::post().and(routes))
+    //     .tls()
+    //     .cert_path("admission-controller-tls.crt")
+    //     .key_path("admission-controller-tls.key")
+    //     .run(([0, 0, 0, 0], 8443))
+    //     .await;
+
+    warp::serve(warp::post().and(routes))
+        .run(([0, 0, 0, 0], 8080))
+        .await;
+}
+
+async fn mutate(body: AdmissionReview<DynamicObject>) -> Result<impl Reply, Infallible> {
+    let mut res: AdmissionResponse = Default::default();
+
+    let req: AdmissionRequest<DynamicObject> = match body.try_into() {
+        Ok(req) => req,
+        Err(err) => {
+            res.allowed = false;
+            res.result.message = Some(err.to_string());
+
+            error!("invalid request: {}", err);
+            return Ok(reply::json(&res.into_review()));
```

Comment:
This feels like a `reply::json(&res.deny(reason).into_review())`.

I.e. `AdmissionResponse` can have builders, and `into_review` is the finaliser of it.